packages/firestore/src/util/async_queue.ts

```diff
@@ -240,7 +240,7 @@ export class AsyncQueue {
 
   constructor() {
     const window = PlatformSupport.getPlatform().window;
-    if (window) {
+    if (window && typeof window.addEventListener === 'function') {
```

Comment:
Why checking for the type particularly? Are there platforms where this exists but not a function? Just curious, LGTM anyways.

Reply:
I don't want to find out....

Reply:
This check is needed for React Native dev.